src/app/services/baw-api/api-common.ts

```diff
@@ -83,30 +91,46 @@ export interface ApiUpdate<M extends AbstractModel, P extends any[]> {
 /**
  * API Delete functionality
  */
-export interface ApiDestroy<M extends AbstractModel, P extends any[], I extends IdOr<M>> {
+export interface ApiDestroy<
+  M extends AbstractModel,
+  P extends any[],
+  I extends IdOr<M>
+> {
   /**
    * destroy  individual model
    * @param args URL parameter values
    */
-  destroy(model: I, ...urlParameters: P): Observable<M>;
+  destroy(model: I, ...urlParameters: P): Observable<null>;
 }
 
 /**
  * Api Class with all abilities enabled
  */
-export abstract class StandardApi<
-  M extends AbstractModel,
-  P extends any[]> extends BawApiService<M> implements
-  ApiList<M, P>,
-  ApiFilter<M, P>,
-  ApiShow<M, P, IdOr<M>>,
-  ApiCreate<M, P>,
-  ApiUpdate<M, P>,
-  ApiDestroy<M, P, IdOr<M>> {
+export abstract class StandardApi<M extends AbstractModel, P extends any[]>
+  extends BawApiService<M>
+  implements
+    ApiList<M, P>,
+    ApiFilter<M, P>,
+    ApiShow<M, P, IdOr<M>>,
+    ApiCreate<M, P>,
+    ApiUpdate<M, P>,
+    ApiDestroy<M, P, IdOr<M>> {
   abstract list(...urlParameters: P): Observable<M[]>;
   abstract filter(filters: Filters, ...urlParameters: P): Observable<M[]>;
   abstract show(model: IdOr<M>, ...urlParameters: P): Observable<M>;
   abstract create(model: M, ...urlParameters: P): Observable<M>;
-  abstract update(model: PartialWith<M, "id">, ...urlParameters: P): Observable<M>;
-  abstract destroy(model: IdOr<M>, ...urlParameters: P): Observable<M>;
+  abstract update(
+    model: PartialWith<M, "id">,
+    ...urlParameters: P
+  ): Observable<M>;
+  abstract destroy(model: IdOr<M>, ...urlParameters: P): Observable<null>;
+}
+
+/**
+ * Api class with only the show route enabled
+ */
+export abstract class ShowApi<M extends AbstractModel, P extends any[]>
```

Comment:
So you probably don't need to create a child abstract class for every combination of API methods that may be implemented in a service. It was my intention that one would just implement the interface they needed.

The abstract class here is there really to just combine a whole wack of type signatures into a nice bundle.

Still, if you find this construct useful - go for it.